src/platform/silabs/efr32/BLEManagerImpl.cpp

```diff
@@ -950,11 +968,23 @@ uint8_t BLEManagerImpl::GetTimerHandle(uint8_t connectionHandle, bool allocate)
 
 void BLEManagerImpl::BleAdvTimeoutHandler(TimerHandle_t xTimer)
 {
-    if (BLEMgrImpl().mFlags.Has(Flags::kFastAdvertisingEnabled))
-    {
+    if (BLEMgrImpl().mFlags.Has(Flags::kFastAdvertisingEnabled)) {
         ChipLogDetail(DeviceLayer, "bleAdv Timeout : Start slow advertissment");
+        BLEMgrImpl().mFlags.Set(Flags::kAdvertising);
```

Comment:
That is a very good question. Not all platforms use the same priority model as your examples, and depending on that this might be necessary, since SetAdvertisingMode triggers delayed execution of the StartAdvertising. 

If we are already advertising it does not matter and it covers a wider range of cases.